### PR TITLE
feat(RHIENG-10400): migrate policy systems table to REST

### PR DESCRIPTION
**Pull request by mkholjuraev**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

This migrates the Policy systems table to Rest API. Due to API shortcomings, ssgVersion column has been hidden. Also, all the disabled functionalities in SystemsTable component applies to this page as well. They are selection, tag filter.

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by LightOfHeaven1994 — COMMENTED** [review]
We are also missing:

1. Tags filtering (searching by tags)
2. Select all (just wanted to mention, I know it's going to be covered later)

**Comment by mkholjuraev** [comment]
@LightOfHeaven1994 Thanks for the initial review. Yes,  as I mentioned in the PR description, tag filtering, selection are disabled. Regarding the OS endpoint issue, if I understand correctly, each page should use different endpoints for fetching OS versions right?

**Comment by LightOfHeaven1994** [comment]
> @LightOfHeaven1994 Thanks for the initial review. Yes, as I mentioned in the PR description, tag filtering, selection are disabled. Regarding the OS endpoint issue, if I understand correctly, each page should use different endpoints for fetching OS versions right?

yes, each endpoint has it's own /os_versions . It's visible in the APIv2 doc

**Comment by mkholjuraev** [comment]
Needs to be reviewed after: https://github.com/RedHatInsights/insights-inventory-frontend/pull/2262

File: src/Utilities/hooks/useAPIV2FeatureFlag.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import useFeatureFlag from './useFeatureFlag';
 
-const useAPIV2FeatureFlag = () => useFeatureFlag('compliance-api-v2');
```
Comment (AsToNlele) [inline src/Utilities/hooks/useAPIV2FeatureFlag.js:3]: Let's not forget to revert this

File: src/Utilities/hooks/api/useOperatingSystems.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import useQuery, { apiInstance } from '../useQuery';
 
 const useOperatingSystemsQuery = (filters) => {
-  return useQuery(apiInstance.systemsOS, { params: [undefined, filters] });
+  return useQuery(apiInstance.systemsOS, { params: [null, filters] });
```
[the remark this answers is not in the document]
Comment (mkholjuraev) [reply src/Utilities/hooks/api/useOperatingSystems.js:4]: @LightOfHeaven1994 as of the last commit, we are using proper endpoints to fetch operating systems.

File: src/Utilities/hooks/useAPIV2FeatureFlag.js (hunk 1 above)
Comment (mkholjuraev) [reply src/Utilities/hooks/useAPIV2FeatureFlag.js:3]: ah, very true :)

Comment (mkholjuraev) [reply src/Utilities/hooks/useAPIV2FeatureFlag.js]: Done!

**Review by LightOfHeaven1994 — COMMENTED** [review]
When we are sorting by os_version on Policy details Systems page I see that we are sending `sort_by=os_major_version=X and sort_by=os_minor-version=Y` - we don't need that on that page since major version is constant. We need to sort only by os_minor version here

image

**Comment by mkholjuraev** [comment]
@LightOfHeaven1994 

> we don't need that on that page since major version is constant. We need to sort only by os_minor version here

As we share the same code under the hood for several pages and this does not affect the actual behaviour, I would suggest not to add additional conditions/complexity to the sorting. It is okay to have both major and minor versions passed into request. What do you think?

**Comment by LightOfHeaven1994** [comment]
> @LightOfHeaven1994
> 
> > we don't need that on that page since major version is constant. We need to sort only by os_minor version here
> 
> As we share the same code under the hood for several pages and this does not affect the actual behaviour, I would suggest not to add additional conditions/complexity to the sorting. It is okay to have both major and minor versions passed into request. What do you think?

agree, let's leave it like this for now. We can always return to it later.

**Review by LightOfHeaven1994 — APPROVED** [review]
I think we are good now, great work!

**Review by AsToNlele — COMMENTED** [review]
We're missing empty state when the policy has no systems.

Rest API
image

GraphQL
image

Other than that well done!

**Comment by mkholjuraev** [comment]
@AsToNlele I am aware of the issue and it is tricky to resolve due to the difference in the API results. I am happy to explain/discuss it at the standup.

**Comment by mkholjuraev** [comment]
@AsToNlele update to my prev comment:

Actually, the empty state is not shown in both stage with GraphQL as well. You saw the empty state because the policy had no added systems at all. In this case, the empty state with the warning is shown. In case, policy has at least one system and there is an active filter that results in an empty state, then the table would not have an empty state in it even before the migration. Thus, this is not a regression.

**Pull request merged**